File: Cpp/src/objects/output/Mention.cpp
#include "Mention.h"


Mention::Mention() {
    locations = new vector<Location*>();
}

Mention::~Mention() {
    delete locations;
}

void Mention::Serialize(Json::Value& root) {}

void Mention::Deserialize(Json::Value& root) {
    label = root.get("label", "").asString();

    is_negated = root.get("is_negated", true).asBool();
    negating_phrase = root.get("negating_phrase", "").asString();

    // Locations
    if (NULL == this->locations) {
        this->locations = new vector<Location*>();
    }

    Json::Value locations = root["locations"];
    for ( unsigned int i = 0; i < locations.size(); ++i ) {
        Location* location = new Location();
        location->Deserialize(locations[i]);
        this->locations->push_back(location);
    }
}

File: Java/src/main/java/com/semantria/mapping/output/Mention.java
package com.semantria.mapping.output;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlElementWrapper;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Mention
{
	private String label = null;
	private Boolean is_negated = null;
	private String negating_phrase = null;
	private List<Location> locations = null;

	public Mention() { }

	public Mention(String label, boolean is_negated, String negating_phrase, Location location)
	{
		this.label = label;
		this.is_negated = is_negated;
		this.negating_phrase = negating_phrase;
		this.locations = new ArrayList<Location>( Arrays.asList(location) );
	}

	@XmlElement(name = "label")
	public String getLabel() { return label; }
	@XmlElement(name = "is_negated")
	public Boolean getIsNegated() { return is_negated; }
	@XmlElement(name = "negating_phrase")
	public String getNegatingPhrase() { return negating_phrase; }
	@XmlElementWrapper(name = "locations")
	@XmlElement(name = "location")
	public List<Location> getLocations() { return locations; }

	public void setLabel(String label) { this.label = label; }
	public void setIsNegated(Boolean is_negated) { this.is_negated = is_negated; }
	public void setNegatingPhrase(String negating_phrase) { this.negating_phrase = negating_phrase; }
	public void setLocations(List<Location> locations) { this.locations = locations; }
}
